### ISSUE TITLE
[Feature] Export Data Based on Backup S3 Bucket

### ISSUE THREAD

**Issue by hanahmily** [apache/skywalking#12815]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

#### Description

This feature relies on the **Backup Agent** functionality outlined in #12876. The backup agent transfers the data to a S3 bucket, the **Export Command LIne** will enable exporting data based on this bucket, allowing users to extract and utilize specific sets of data efficiently.

#### Requirements

1. **Integration with Backup Snapshots**:
   - Leverage the S3 bucket created by the **Backup Agent** feature as the data source for export.
   - Ensure the export process clean up the snapshot is created for the exportation.

2. **Export Configuration**:
   - **Time Range Export**: Allow users to define a specific time range from the snapshot for export.
   - **Query-Based Export**: Allow users to filter and export data using specific query criteria (e.g., tags, metrics, identifiers).

3. **Export Formats**:
   - Support commonly used data formats like **JSON** and **CSV**.

### Are you willing to submit a pull request to implement this on your own?

- [ ] Yes I am willing to submit a pull request on my own!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment]
Is Export Manager going to be a separate role node?

**Comment by hanahmily** [comment]
> Is Export Manager going to be a separate role node?

It is a command-line tool provided by `bydbctl`, which can be scheduled to run automatically using `cron` on a daily basis or at specified intervals.

**Comment by wu-sheng** [comment]
So, the tool is going to grap snapshots from mutiple data nodes(through liaison)?
Where the filter happens? Are they processed on data nodes?

**Comment by wu-sheng** [comment]
About the feature, the exportation should support tags/fields selection if possible. This could reduce the volume costs of the files.

**Comment by hanahmily** [comment]
I want to create a design to explain the details. Perhaps we can set up exporting to the backup process.

**Comment by wu-sheng** [comment]
Exporting could be a separate tool, and it is not critical for the whole ecosystem. 
This tool will read the whole snapshot from the s3 or efs, and filter out the data to generate exporting result.
We will reserve this for new developer programs, such as GSOC or OSPP 2025

**Comment by prashanth9968** [comment]
Hi @wu-sheng and @hanahmily! I am a student who is good at java with microservices architecture,backend development and dealing with strong formats.I see you are reserving this feature task for new developer programs like GSoC.I am very keen on participating and would love to start discussing the design details for this S3 exporting tool under your guidance.May i pick this up?

**Comment by wu-sheng** [comment]
> Hi @wu-sheng and @hanahmily! I am a student who is good at java with microservices architecture,backend development and dealing with strong formats.I see you are reserving this feature task for new developer programs like GSoC.I am very keen on participating and would love to start discussing the design details for this S3 exporting tool under your guidance.May i pick this up?

This issue targets to the BanyanDB, which is a Go DB, not a Java.

**Comment by SammyUrfen** [comment]
Hi @hanahmily @wu-sheng, is this still held for GSoC or OSPP? The last word on that is from February 2025.

@hanahmily, you mentioned wanting to write the design first. Is that still yours to write, or would a proposal from me be more useful to react to?

Happy to write one if it helps. If the issue is still spoken for, I will leave it be.

**Comment by hanahmily** [comment]
> Hi [@hanahmily](https://github.com/hanahmily) [@wu-sheng](https://github.com/wu-sheng), is this still held for GSoC or OSPP? The last word on that is from February 2025.
> 

> [@hanahmily](https://github.com/hanahmily), you mentioned wanting to write the design first. Is that still yours to write, or would a proposal from me be more useful to react to?
> 
> Happy to write one if it helps. If the issue is still spoken for, I will leave it be.

No, it's not. @mrproliu is working on it. His design spec has passed the review phase. You can check out other issues. Thank you

**Comment by mrproliu** [comment]
I have [created a new one to track](https://github.com/apache/skywalking/issues/14102), so this issue has been closed.
